src/TorchSharp/Tensor/Factories/tensor_Complex.cs

```diff
@@ -271,5 +271,14 @@ public static Tensor tensor(System.Numerics.Complex[,,,] rawArray, ScalarType? d
         {
             return _tensor_generic(rawArray, stackalloc long[] { rawArray.GetLongLength(0), rawArray.GetLongLength(1), rawArray.GetLongLength(2), rawArray.GetLongLength(3) }, (sbyte)ScalarType.ComplexFloat64, dtype, device, requires_grad, names: names);
         }
+
+        /// <summary>
+        /// Create a tensor from an array of values, shaping it based on the shape passed in.
+        /// </summary>
+        [Pure]
```

Comment:
What's `Pure` and what's the purpose of using `Pure` here.

Trying search the `Pure` attribution on MS learn

```
Indicates that a type or method is pure, that is, it does not make any visible state changes.
```

Which really doesn't help a lot here...

Reply:
Well, no, but it's been systematically used on methods throughout the API.